libcudacxx/include/cuda/std/__cuda/atomic.h

```diff
@@ -23,229 +25,86 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
-using std::__detail::thread_scope;
-using std::__detail::thread_scope_block;
-using std::__detail::thread_scope_device;
-using std::__detail::thread_scope_system;
-using std::__detail::thread_scope_thread;
-
-namespace __detail
-{
-using std::__detail::__thread_scope_block_tag;
-using std::__detail::__thread_scope_device_tag;
-using std::__detail::__thread_scope_system_tag;
-} // namespace __detail
-
-using memory_order = std::memory_order;
-
-constexpr memory_order memory_order_relaxed = std::memory_order_relaxed;
-constexpr memory_order memory_order_consume = std::memory_order_consume;
-constexpr memory_order memory_order_acquire = std::memory_order_acquire;
-constexpr memory_order memory_order_release = std::memory_order_release;
-constexpr memory_order memory_order_acq_rel = std::memory_order_acq_rel;
-constexpr memory_order memory_order_seq_cst = std::memory_order_seq_cst;
-
 // atomic<T>
 
 template <class _Tp, thread_scope _Sco = thread_scope::thread_scope_system>
-struct atomic : public std::__atomic_base<_Tp, _Sco>
+struct atomic : public std::__atomic_impl<_Tp, _Sco>
```

Comment:
To be sure, we do want to inherit from `std::atomic`? If so, then all occurences of `std::` need to be changed to `::std::`

Reply:
I think you mean our fully qualified name. We're definitely not inheriting from std::atomic.